### PR TITLE
1807 Add schema registry security params

### LINKED ISSUE #1807: Use authenticated user when sending _INFO to Kafka
## Background
In case Kafka requires user authentication to write into a topic, this needs to be supported by the Kafka plugin

## Feature
It's to be discussed and decided if the user executing the `SparkJobs` would be the one authenticating against Kafka, or there would be one user set-up to do the writing. Or the former being a fallback of the later.

### PR DESCRIPTION
**Pull request by Zejnilovic**
New configs for SparkJobs using kafak plugins
- kafka.schema.registry.basic.auth.credentials.source
- kafka.schema.registry.basic.auth.user.info

Release Notes:
Added new properties for KafkaPlugins to be able to connect to Secured SchemaRegistry

Fixes #1807

### CODE REVIEW

File: plugins-builtin/src/main/scala/za/co/absa/enceladus/plugins/builtin/common/mq/kafka/KafkaSecurityParams.scala [hunk 1]
```diff
@@ -27,8 +27,12 @@ case class KafkaSecurityParams(
...
                               ) {
   def toMap: Map[String, String] = {
-    Map(SecurityProtocolKey -> securityProtocol) ++
-      saslMechanism.fold(Map.empty[String, String])(saslValue => Map(SaslMechanismKey -> saslValue))
+    Map(
+      SecurityProtocolKey -> Some(securityProtocol),
+      SaslMechanismKey -> saslMechanism
+    ).collect {
+      case (key, Some(value)) => key -> value
+    }
```
Comment (Zejnilovic) [inline plugins-builtin/src/main/scala/za/co/absa/enceladus/plugins/builtin/common/mq/kafka/KafkaSecurityParams.scala:35]: Did this in the first iteration when I added 3 more params. I would just leave it be if no one has any comments.

Comment (AdrianOlosutean) [reply plugins-builtin/src/main/scala/za/co/absa/enceladus/plugins/builtin/common/mq/kafka/KafkaSecurityParams.scala:35]: There is also this option to use:
`for ((k, Some(v)) <- map) yield k -> v`
But I would not say it's that much better

**Review by AdrianOlosutean — APPROVED** [review]
Looks good so far, Im wondering where to test it

File: plugins-builtin/src/main/scala/za/co/absa/enceladus/plugins/builtin/common/mq/kafka/SchemaRegistrySecurityParams.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+case class SchemaRegistrySecurityParams(credentialsSource: String,
+                                        userInfo: Option[String])
+
+object SchemaRegistrySecurityParams {
+  val BasicAuthCredentialsSourceKey = "kafka.schema.registry.basic.auth.credentials.source"
+  val BasicAuthUserInfoKey = "kafka.schema.registry.basic.auth.user.info"
+
+  def fromConfig(conf: Config): Option[SchemaRegistrySecurityParams] = {
+    conf.getOptionString(BasicAuthCredentialsSourceKey).map { source =>
```
Comment (benedeki) [inline plugins-builtin/src/main/scala/za/co/absa/enceladus/plugins/builtin/common/mq/kafka/SchemaRegistrySecurityParams.scala:29]: This is neat! 😎

File: plugins-builtin/src/main/scala/za/co/absa/enceladus/plugins/builtin/common/mq/kafka/KafkaSecurityParams.scala (hunk 1 above)
Comment (benedeki) [reply plugins-builtin/src/main/scala/za/co/absa/enceladus/plugins/builtin/common/mq/kafka/KafkaSecurityParams.scala:35]: I think it's OK , how it is.

**Review by benedeki — APPROVED** [review]
- [x] code reviewed
- [x] pulled
- [x] built
- [ ] run

**Comment by dk1844** [comment]
Test-ran on Chewbacca, the info plugin actually wrote data to a Kafka topic while using configuration such as
```properties
kafka.schema.registry.basic.auth.credentials.source=USER_INFO
kafka.schema.registry.basic.auth.user.info="SVC-user-accounT:secretPasswordExampleHere#2+a"
```

**Pull request merged**